pkg/utils/netlink.go

```diff
@@ -108,3 +119,40 @@ func BridgeTrunkVlanAdd(nlink Netlink, link netlink.Link, vlans []int) error {
 	}
 	return nil
 }
+
+// GetBridgeByLink returns linux bridge link if provided link belongs to any
+// if provided link is a part of a bond will return a bridge to which this bond belongs to
+func GetBridgeByLink(nLink Netlink, link netlink.Link) (netlink.Link, error) {
+	master, err := getMasterInterface(nLink, link)
+	if err != nil {
+		return nil, err
+	}
+	if master == nil {
+		return nil, nil
+	}
+	switch master.Type() {
```

Comment:
should we search for the link which contains bridge in a generic manner  ?

e.g while loop that checks for master until it hits a master of type bridge then returns that bridge

```
master := link
while master.Type() != linkTypeBridge {
    master, err = getMasterInterface(nLink, master)
    if err != nil || master == nil {
    // dead end
    return nil, fmt.Errorf("error....")
    }
}
return master
```

Reply:
ok, let's use proposed snippet for bridge detection